pilot/pkg/bootstrap/server.go

```diff
@@ -336,6 +338,7 @@ func (s *Server) Start(stop <-chan struct{}) error {
 				return
 			}
 			log.Infof("starting secure gRPC discovery service at %s", s.SecureGrpcListener.Addr())
+			reflection.Register(s.secureGrpcServer)
```

Comment:
Reflection moved here to avoid multiple calls to Register, in case of multiplexing ( panic otherwise)

Reply:
I don't really understand why the reflect cal would panic? If it panics for duplicate register, shouldn't the same thing happen for duplicate registering ADS?

Reply:
It seems gRPC library doesn't like if you call reflection.Register 2x on the same server - or it did for me. 
I did some other refactorings after - so it may not be needed anymore. Let me check.

Reply:
Yes, not needed, reverted.